src/GraphQL/ParseGraphQLServer.js

```diff
@@ -57,8 +58,21 @@ class ParseGraphQLServer {
     if (schemaRef === newSchemaRef && this._server) {
       return this._server;
     }
-    const options = await this._getGraphQLOptions();
-    this._server = createServer(options);
+    const { schema, context } = await this._getGraphQLOptions();
+    const apollo = new ApolloServer({
+      csrfPrevention: {
+        // See https://www.apollographql.com/docs/router/configuration/csrf/
+        // needed since we use graphql upload
+        requestHeaders: ['X-Parse-Application-Id'],
+      },
+      introspection: true,
```

Comment:
Introspection should probably be disabled by default for security reasons. Perhaps a config variable.

Reply:
Hi @kekami , you are right but it needs a dedicated PR, i created the FR: https://github.com/parse-community/parse-server/issues/8800